### PR TITLE
GH-159

### PR DESCRIPTION
**Pull request by evdhoudt**
Closes NCI-Agency/anet#159

### CODE REVIEW

File: client/src/components/AdvisorReports/FilterableAdvisorReportsTable.js [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+
+    handleFilterTextInput(filterText) {
+        this.setState({ filterText: filterText })
+    }
+
+    handleExportButtonClick() {
+        let selectedData = this.state.selectedData
+        let allData = this.state.data
+        let exportData = (selectedData.length > 0) ? selectedData : allData
+        console.log(this.convertArrayOfObjectsToCSV({ data: exportData }))
```
Comment (maradragan) [inline client/src/components/AdvisorReports/FilterableAdvisorReportsTable.js]: You have a debug statement in here.

File: client/src/components/AdvisorReports/AdvisorReportsModal.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
+import React, { Component } from 'react'
+import SimpleModal from 'components/SimpleModal'
+import AdvisorReportsTable from 'components/AdvisorReports/AdvisorReportsTable'
+
+import API from 'api'
+const advisorsQueryUrl = `/api/reports/insights/advisors` // ?weeksAgo=3 default set at 3 weeks ago
+
+
+class AdvisorReportsModal extends Component {
+    constructor(props) {
```
Comment (maradragan) [inline client/src/components/AdvisorReports/AdvisorReportsModal.js]: I would add here the propTypes.

File: client/src/components/AdvisorReports/AdvisorReportsRow.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+        else {
+            stats.push(<td key={ keySubmitted }>0</td>)
+            stats.push(<td key={ keyAttended }>0</td>)
+        }
+    })
+    return stats
+}
+
+const AdvisorReportsRow = (props) => {
+    let statistics = _advisorStats(props.columnGroups, props.row.stats)
```
Comment (maradragan) [inline client/src/components/AdvisorReports/AdvisorReportsRow.js]: I would add here the propTypes. That way you force certain property types and you also document the component better.

File: client/src/components/AdvisorReports/AdvisorReportsTable.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
+import React from 'react'
+import { Table } from 'react-bootstrap'
+import AdvisorReportsTableHead from 'components/AdvisorReports/AdvisorReportsTableHead'
+import AdvisorReportsRow from 'components/AdvisorReports/AdvisorReportsRow'
+import _uniqueId from 'lodash.uniqueid'
+
+const AdvisorReportsTable = (props) => {
+    let rows = props.data.map( (advisor) => {
```
Comment (maradragan) [inline client/src/components/AdvisorReports/AdvisorReportsTable.js]: I would add here the propTypes. That way you force certain property types and you also document the component better.

File: client/src/components/AdvisorReports/AdvisorReportsTableHead.js [hunk 5]
```diff
@@ -0,0 +1,35 @@
+import React from 'react'
+import Checkbox from 'components/Checkbox'
+
+const AdvisorReportsTableHead = (props) => {
+    let weekHeadings = []
```
Comment (maradragan) [inline client/src/components/AdvisorReports/AdvisorReportsTableHead.js]: I would add here the propTypes. That way you force certain property types and you also document the component better.

File: client/src/components/AdvisorReports/FilterableAdvisorReportsTable.js [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+import _debounce from 'lodash.debounce'
+import moment from 'moment'
+
+import API from 'api'
+
+const DEFAULT_WEEKS_AGO = 3
+const advisorReportsQueryUrl = `/api/reports/insights/advisors` // ?weeksAgo=3 default set at 3 weeks ago
+
+class FilterableAdvisorReportsTable extends Component {
+    constructor(props) {
```
Comment (maradragan) [inline client/src/components/AdvisorReports/FilterableAdvisorReportsTable.js]: I would add here the propTypes. That way you force certain property types and you also document the component better.

File: client/src/components/AdvisorReports/Toolbar.js [hunk 7]
```diff
@@ -0,0 +1,40 @@
+import React, { Component } from 'react'
+import { Button } from 'react-bootstrap'
+
+class Toolbar extends Component {
+    constructor(props) {
```
Comment (maradragan) [inline client/src/components/AdvisorReports/Toolbar.js]: I would add here the propTypes. That way you force certain property types and you also document the component better.

File: client/src/components/SimpleModal.js [hunk 8]
```diff
@@ -0,0 +1,46 @@
+import React, { Component } from 'react'
+import { Button, Modal } from 'react-bootstrap'
+
+class SimpleModal extends Component {
+    constructor(props){
```
Comment (maradragan) [inline client/src/components/SimpleModal.js]: I would add here the propTypes. That way you force certain property types and you also document the component better.

File: src/main/java/mil/dds/anet/utils/Utils.java [hunk 9]
```diff
@@ -163,5 +165,76 @@ public static DateTime handleRelativeDate(DateTime input) {
...
+	 * </code>
+	 * then transforming it like so:
+	 * <code>
+	 * final Set<String> tlf = Stream.of("name", "someprop").collect(Collectors.toSet());
+	 * return resultGrouper(list, "stats", "id", tlf);
+	 * </code>
+	 * will return the list:
+	 * <code>
+	 * [{"id": 3, "name": "alice", "someprop": "x",
+	 *    [{"week": 39, "nrdone": 2, "nrtodo": 0},
```
Comment (maradragan) [inline src/main/java/mil/dds/anet/utils/Utils.java]: I think here misses the groupname key.

**Comment by evdhoudt** [comment]
I've addressed all the issues and pushed new commits to fix them.

File: client/package.json [hunk 10]
```diff
@@ -83,6 +83,7 @@
     "lodash.last": "^3.0.0",
     "lodash.mapvalues": "^4.6.0",
     "lodash.some": "^4.6.0",
+    "lodash.uniqueid": "^4.0.1",
```
Comment (gjvoosten) [inline client/package.json]: Please merge latest candidate, then freeze this dependency at version 4.0.1.

**Pull request merged**